### PR TITLE
CASMPET- 6520 Add Vault support to TAPMS

### PR DESCRIPTION
**Pull request by rnoska-hpe**
## Summary and Scope

This change allows the Cray Tenant TAPMS operator to maintain Cray Vault transit engines for tenants. The change updates the cray-tapms-crd Helm chart to extend the TAPMS CRD schema and updates the cray-tapms-operator chart to implement support for Vault. 

## Issues and Related PRs

* Resolves [CASMPET- 6520](https://jira-pro.it.hpe.com:8443/browse/CASMPET-6520) and several other items in the master [CASM-3945](https://jira-pro.it.hpe.com:8443/browse/CASM-3945) epic for this feature.

## Testing

The cray-tapms-crd and cray-tapms-operator Helm charts were upgraded on Ashton and basic tests were run to confirm that the required Vault functionality works as expected. The actual tests are documented [here](https://rndwiki-pro.its.hpecorp.net/display/CASMPET/PET+Multi-Tenancy+Phase+2+Test+Plan#PETMultiTenancyPhase2TestPlan-VaultTransitIntegration)

### Tested on:

  * Ashton (VShasta)

### Test description:

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? YES. 
- Was downgrade tested? YES
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

None. This is for customer preview and not in use yet.

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by bklei — APPROVED** [review]

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,28 +1,3 @@
-/*
```
Comment (jeremy-duckworth) [inline api/v1alpha1/zz_generated.deepcopy.go:1]: Should the copyright be in place here?

File: api/v1alpha2/tenant_types.go [hunk 2]
```diff
@@ -52,16 +52,49 @@ type TenantResource struct {
...
+	KeyType string `json:"keytype,omitempty"`
+	// The Vault public key.
+	PublicKey string `json:"publickey,omitempty"`
+}
+
 // @Description The desired state of Tenant
 type TenantSpec struct {
-	TenantName      string   `json:"tenantname" example:"vcluster-blue" binding:"required"`
+	TenantName string `json:"tenantname" example:"vcluster-blue" binding:"required"`
+	//+kubebuilder:validation:Optional
```
Comment (jeremy-duckworth) [inline api/v1alpha2/tenant_types.go:91]: Should the kubebuilder annotation be optional for State here? Not sure this should be optional.

File: kubernetes/cray-tapms-crd/files/tapms.hpe.com_tenants.yaml [hunk 3]
```diff
@@ -158,6 +158,25 @@ spec:
                 type: array
               state:
                 type: string
+              tenantkms:
```
Comment (jeremy-duckworth) [inline kubernetes/cray-tapms-crd/files/tapms.hpe.com_tenants.yaml:161]: A note for future consideration. We might want to move the KMS section under tenantresources and allow multiple KMSs.

**Review by jeremy-duckworth — APPROVED** [review]

File: api/v1alpha1/zz_generated.deepcopy.go (hunk 1 above)
Comment (rnoska-hpe) [reply api/v1alpha1/zz_generated.deepcopy.go:1]: Will address properly in a future commit when I have more time to retest. The issue is that make manifests regenerates this code and drops our copyright.

File: api/v1alpha2/tenant_types.go (hunk 2 above)
Comment (rnoska-hpe) [reply api/v1alpha2/tenant_types.go:91]: This was in error. I will address this in a future commit when I have more time to retest.

File: kubernetes/cray-tapms-crd/files/tapms.hpe.com_tenants.yaml (hunk 3 above)
Comment (rnoska-hpe) [reply kubernetes/cray-tapms-crd/files/tapms.hpe.com_tenants.yaml:161]: I will add a jira item to address this.

**Pull request merged**